src/python/WMCore/WorkQueue/WorkQueue.py

```diff
@@ -1089,19 +1089,8 @@ def processInboundWork(self, inbound_work=None, throw=False, continuous=False):
                         processedInputs = []
                         for unit in work:
                             processedInputs.extend(unit['Inputs'].keys())
-                        if processedInputs or rejectedWork:
-                            chunkSize = 20
-                            chunkProcessed = processedInputs[:chunkSize]
-                            chunkRejected = rejectedWork[:chunkSize]
-                            # TODO:Make this a POST update, instead of several PUT
-                            while processedInputs or rejectedWork:
-                                self.backend.updateInboxElements(inbound.id, ProcessedInputs=chunkProcessed,
-                                                                 RejectedInputs=chunkRejected,
-                                                                 options={'incremental': True})
-                                processedInputs = processedInputs[chunkSize:]
-                                rejectedWork = rejectedWork[chunkSize:]
-                                chunkProcessed = processedInputs[:chunkSize]
-                                chunkRejected = rejectedWork[:chunkSize]
+                        self.backend.updateInboxElements(inbound.id, ProcessedInputs=processedInputs,
+                                                                 RejectedInputs=rejectedWork)
```

Comment:
Cool! What about that "incremental" option?

Reply:
incremental option was needed since update handler was getting the data in request url since everything passed to request body now, there is no need for "incremental" option.